sources/core/Game.cpp

```diff
@@ -707,15 +707,15 @@ void Game::generateUnsync()
     }
 }
 
-void Game::AddLastMessages(std::stringstream& stream)
+void Game::AddLastMessages(QByteArray& data)
 {
-    stream << std::endl;
+    data.append('\n');
     for (int i = (log_pos_ + 1) % messages_log_.size();
              i != log_pos_;
              i = (i + 1) % messages_log_.size())
     {
-        stream << messages_log_[i].type << " ";
-        stream << messages_log_[i].json.toStdString() << std::endl;
+        data.append(messages_log_[i].type + " ");
+        data.append(messages_log_[i].json + '\n');
```

Comment:
The actual type of `" "` is `const char[2]` (constant array of chars, the size is two).
But plain arrays in C++ is actually just constant pointers (there are some differences, but not so much), so string literals in C++ (`"something something"`) in a wide amount of cases behave like constant pointers (like a `const char*` pointer).